api/src/utils/merge-version-data.ts

```diff
@@ -0,0 +1,186 @@
+import type { Item, SchemaOverview } from '@directus/types';
+import Joi from 'joi';
+import type { Alterations } from '../types/index.js';
+
+const alterationSchema = Joi.object({
+	create: Joi.array().items(Joi.object().unknown()),
+	update: Joi.array().items(Joi.object().unknown()),
+	delete: Joi.array().items(Joi.string(), Joi.number()),
+});
+
+export function mergeVersionsRaw(item: Item, versionData: Partial<Item>[]) {
+	const result = { ...item };
+
+	for (const versionRecord of versionData) {
+		for (const key in versionRecord) {
```

Comment:
As this is using `in` should we be checking `Object.prototype.hasOwnProperty.call(versionRecord, key)`?

Reply:
Yes, but preferably via `for (const [key, value] of Object.entries(versionData))`.
Also this applies to other similar usages in the file as well.